### ISSUE TITLE
Is it possible to determine if render is in progress for async paper?

### ISSUE THREAD

**Issue by chipit24** [clientIO/joint#1219]
I have set up [async rendering](https://resources.jointjs.com/docs/jointjs/v3.1/joint.html#dia.Paper.prototype.options.async) on the paper, and perform an action such as calling [`toFront`](https://resources.jointjs.com/docs/jointjs/v3.1/joint.html#dia.Element.prototype.toFront) on an element. If the z-index is actually modified, I notice that a `render:done` event is triggered on the paper, however, if the shape it already at the front, after calling `toFront`, the `render:done` event is not triggered.

Is there some way to determine if the paper is still rendering, or has finished rendering even if no `render:done` event has been triggered? This would be useful in situations like the following:

```js
shape.toFront();

paper.once('render:done', () => {
  // Do work that requires shape to be at front
  // If shape is already at the front, this won't be called ...
});
```

It's a bit of a contrived example but I hope it illustrates a need to determine knowing that rendering wasn't actually triggered, or is still in progress.

**Comment by kumilingus** [comment]
That makes complete sense

Do you use the `viewport` paper option?  I need to think about a proper solution, which will work even when `viewport` matching feature is enabled.  To keep you going, here's a simple method, which should does the trick.

```js
joint.dia.Paper.prototype.hasScheduledUpdates = function() {
    return this._updates.priorities.some(updates => !joint.util.isEmpty(updates));
};
```

**Comment by chipit24** [comment]
I'm not using the `viewport` option, though I didn't know about it until now, so I may end up implementing it at some point 😃

I'll try the code snippet you posted, thanks!!

**Comment by luis-carlos-campos** [comment]
@chipit24 @kumilingus
Am I right to assume that if you are using an async paper (with a custom viewport method) **render:done** would not be called once all valid elements (**viewport rule**) are displayed?

**Comment by kumilingus** [comment]
@luis-carlos-campos sorry for not answering. `render:done` will be called after all views are checked against viewport rule and those who pass rendered.

`hasScheduledUpdates` implemented in `v3.3.`.
